tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -1151,6 +1151,32 @@ void test_oops_stackcheck(void)
 	test_oops(K_ERR_STACK_CHK_FAIL, K_ERR_STACK_CHK_FAIL);
 }
 
+void z_impl_check_syscall_context(void)
+{
+	int key = irq_lock();
+
+	irq_unlock(key);
```

Comment:
shouldn't this unlock be done after `zassert_true(arch_irq_unlocked(key), "irqs locked during syscall");` ?
to test that interruption was not locked ?

Reply:
@ceolin no, it shouldn't.
think about what the key value actually represents.

Reply:
@ceolin the key value has the *previous* locking before irq_lock() was called. this is what we are checking, what the state was when we entered the syscall. the test is not intended to prove that irq_lock locks interrupts.

Reply:
I got what you meant now, thanks for clarifying it.